### PR TITLE
sci-libs/pdal: bump to 2.4.0

### PR DESCRIPTION
**Pull request by t0b3**

### CODE REVIEW

File: sci-libs/pdal/pdal-2.4.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	net-misc/curl
+	sci-libs/gdal
+	sci-libs/libgeotiff
+	sci-geosciences/laszip
+	sys-libs/libunwind
+	sys-libs/zlib
+	postgres? ( dev-db/postgresql:*[xml] )
+	test? ( sci-libs/gdal[geos,jpeg,png] )
+"
+RDEPEND="${DEPEND}"
```
Comment (thesamesam) [inline sci-libs/pdal/pdal-2.4.0.ebuild:32]: The `test? ( ...)` dep doesn't belong in RDEPEND. Consider swapping DEPEND & RDEPEND to solve this, and adding the test dep in DEPEND.

Also, use := on any deps we link against that expose a subslot.

Comment (t0b3) [reply sci-libs/pdal/pdal-2.4.0.ebuild:32]: thank you for this feedback.

File: sci-libs/pdal/pdal-2.4.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	virtual/pkgconfig
+"
+RDEPEND="
+	dev-libs/jsoncpp:=
+	net-misc/curl
+	sci-libs/gdal:=
+	sci-libs/libgeotiff:=
+	sci-geosciences/laszip
+	sys-libs/libunwind:=
+	sys-libs/zlib:=
```
Comment (juippis) [inline sci-libs/pdal/pdal-2.4.0.ebuild:28]:
```suggestion
	sys-libs/zlib
```

File: sci-libs/pdal/pdal-2.4.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+	${RDEPEND}
+"
+
+S="${WORKDIR}/PDAL-${PV}-src"
+
+src_configure() {
+	local mycmakeargs=(
+		-DBUILD_PLUGIN_PGPOINTCLOUD="$(usex postgres)"
+		-DWITH_LAZPERF=OFF
+		-DWITH_LASZIP=ON
```
Comment (juippis) [inline sci-libs/pdal/pdal-2.4.0.ebuild:42]:
```
 * One or more CMake variables were not used by the project:
 *   WITH_LASZIP
 *   WITH_LAZPERF
```

Also leading to:

```
 * VDB: detected possibly incorrect RDEPEND (sci-libs/pdal-2.4.0)
 * dev-db/postgresql:14   < 
 * dev-libs/jsoncpp:=     < 
 *                        > app-arch/zstd:=
 *                        > dev-libs/libxml2
 *                        > dev-libs/openssl:=
 * sci-geosciences/laszip < 
```
(unneeded dep)

**Comment by juippis** [comment]
Also forgot to mention, but you need sys-devel/gettext in BDEPEND.

**Comment by t0b3** [comment]
:+1: hope everything is fine now

File: sci-libs/pdal/pdal-2.4.0.ebuild (hunk 3 above)
Comment (t0b3) [reply sci-libs/pdal/pdal-2.4.0.ebuild]: f.yi.i it seems postgres is still needed:
- -- Found PostgreSQL: /usr/lib64/libpq.so (found version "13.5")  
-  * PostgreSQL PointCloud plugin, read/write PostgreSQL PointCloud objects

**Review by juippis — APPROVED** [review]
Tests failed but otherwise great, thanks!

**Comment by t0b3** [comment]
@juippis tests are already fixed upstream https://github.com/PDAL/PDAL/issues/3733 and will land in 2.4.1

**Comment by t0b3** [comment]
@juippis I just remarked the SUBSLOT would needed to be upgraded to 14 now. I guess this would involve to upgrade to -r1, is that the way to go?

**Comment by thesamesam** [comment]
> @juippis I just remarked the SUBSLOT would needed to be upgraded to 14 now. I guess this would involve to upgrade to -r1, is that the way to go?

yes

**Comment by juippis** [comment]
Yes please. Guess I need to start libabigailing these...

**Pull request closed without merging**